### PR TITLE
✨ Adds ability to not filter CRDS, see ISSUE #373

### LINKED ISSUE #373: support for CRD's that extend common types
##### ISSUE TYPE
 - [ ] Bug Report
 - [X] Feature Idea

##### SUMMARY
I've noticed that Kubeaudit doesn't catch CRDS that are extending typical k8s resources, like apps/v1.
It looks like either with finagling the data out with kubectl and scanning you can still get to the original source of truth.

##### STEPS TO REPRODUCE
create a custom CRD based off apps
scan k8s
don't find anything wrong with deploy applications

##### EXPECTED RESULTS
It should find them

##### other stuff

if you guys want to point me in the direction of where in the code one can specify a custom CRD, that'll work for me and I'll add a flag to support it.

### PR DESCRIPTION
**Pull request by nobletrout**
##### Description
Adds new command flag to not filter out generated resources, which unfortunately also can include CRD based resources (like a CRD based on `Application`)
Fixes #373 

##### Type of change

- [x]  New feature :sparkles:
##### How Has This Been Tested?
I ran it against my cluster and compared results with the flag on and off and found  a lot more issues with the flag, which was my intent.

##### Checklist:

- [x] I have :tophat: my changes (A 🎩 specifically includes pulling down changes, setting them up, and manually testing the changed features and potential side effects to make sure nothing is broken)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] The test coverage did not decrease
- [x] I have signed the appropriate [Contributor License Agreement](https://cla.shopify.com/)

It is kind of difficult to test if these changes work or not, since they only apply to querying remote clusters and not local files. If you guys have a suggestion for me to add test cases, I'm happy to help or learn.

### CODE REVIEW

File: internal/k8sinternal/client.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ func IsRunningInCluster(client Client) bool {
 type ClientOptions struct {
 	// Namespace filters resources by namespace. Defaults to all namespaces.
 	Namespace string
+	IncludeGenerated bool
```
Comment (genevieveluyt) [inline internal/k8sinternal/client.go:88]: Could you add a comment here explaining what the field does so it'll end up in the Go docs?

File: cmd/commands/root.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ type rootFlags struct {
 	namespace   string
 	minSeverity string
 	exitCode    int
+	includegenerated bool
```
Comment (genevieveluyt) [inline cmd/commands/root.go:26]: Can we make this camel case to be consistent?
```suggestion
	includeGenerated bool
```

**Review by genevieveluyt — COMMENTED** [review]
Thanks for your PR! Could you please document the new flag in the [README](https://github.com/Shopify/kubeaudit/blob/main/README.md#global-flags). Perhaps something like:

"By default, kubeaudit will ignore generated resources (such as Pods generated by Deployments). If you would like kubeaudit to produce results for generated resources (for example if you have custom resources or want to catch orphaned resources where the owner resource no longer exists) you can use this flag."

**Comment by genevieveluyt** [comment]
Perfect thanks! As for testing, I can look into adding a test next week.

If you'd like to try it out yourself, I'm thinking of doing something similar to https://github.com/Shopify/kubeaudit/blob/main/internal/test/test.go#L111-L114 with a new test fixture and then seeing if the report results contains the pods or not (or even just comparing the number of results). You can use [`make test-setup`](https://github.com/Shopify/kubeaudit/blob/main/Makefile#L39) to set up a test cluster using kind so you can run the individual test.

But definitely don't feel like you need to do all that, as I said I'm happy to add the testing portion.

**Comment by nobletrout** [comment]
I'll try to get the unit tests to work, but if you beat me to it go ahead. I'm trying to figure out how to not run ALL the unit tests everytime.
any advice?

**Review by genevieveluyt — APPROVED** [review]
Ok I wrote out a test locally and it passes 🎉 . Since I can't push to your branch let's get your PR merged and I'll add the test as a separate PR.

**Pull request merged**